### PR TITLE
chore: 1-click for application forking service

### PR DESCRIPTION
**Pull request by nsarupr**
#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/27075

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationForkingServiceCEImpl.java [hunk 1]
```diff
@@ -29,6 +30,7 @@
 
 @RequiredArgsConstructor
 @Slf4j
+@Component
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationForkingServiceCEImpl.java:33]: Let's keep the annotation consistent, please replace with `@Service` as per `ApplicationForkingServiceImpl`

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce_compatible/ApplicationForkingServiceCeCompatibleImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import com.appsmith.server.services.SessionUserService;
+import com.appsmith.server.services.WorkspaceService;
+import com.appsmith.server.solutions.ApplicationPermission;
+import com.appsmith.server.solutions.ForkExamplesWorkspace;
+import com.appsmith.server.solutions.ImportExportApplicationService;
+import com.appsmith.server.solutions.WorkspacePermission;
+import com.appsmith.server.solutions.ce.ApplicationForkingServiceCEImpl;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce_compatible/ApplicationForkingServiceCeCompatibleImpl.java:15]: Same here

**Review by abhvsn — COMMENTED** [review]
Very minor comment!

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationForkingServiceCEImpl.java (hunk 1 above)
Comment (nsarupr) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationForkingServiceCEImpl.java:33]: This is actually not named correctly. If you see, this is present in the solution folder, and hence I annotated it with @Component

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce_compatible/ApplicationForkingServiceCeCompatibleImpl.java (hunk 2 above)
Comment (nsarupr) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce_compatible/ApplicationForkingServiceCeCompatibleImpl.java:15]: same as above.

**Review by abhvsn — APPROVED** [review]

**Comment by nsarupr** [comment]
Post confirmation from @Aishwarya-U-R  [here](https://theappsmith.slack.com/archives/CGBPVEJ5C/p1694427702323559?thread_ts=1694422627.370609&cid=CGBPVEJ5C). Hence merging the feature branch.

**Pull request merged**